src/Controller/Component/PrgComponent.php

```diff
@@ -46,9 +46,10 @@ public function conversion($redirect = true)
             return null;
         }
         if ($redirect) {
-            return $this->_registry->getController()->redirect(
-                $this->request->params['pass'] + ['?' => $this->request->data]
-            );
+            $parsedUrl = parse_url($this->request->here(false));
```

Comment:
Parsing URL again to remove querystring could be avoided by using `'/' . $this->request->url` instead of `$this->request->here(false)` without would return same URL sans the querystring.